File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        match = {']': '[', ')': '(', '}': '{'}

        for char in s:
            if char in "({[":
                stack.append(char)
            else:
                if not stack or stack[-1] != match[char]:
                    return False
                else:
                    stack.pop()
        
        return len(stack) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.swift
class Solution {
    func isValid(_ s: String) -> Bool {
        var stack: [Character] = []
        let matching: [Character: Character] = [")": "(", "}": "{", "]": "["]

        for char in s {
            if "({[".contains(char) {
                stack.append(char)
            } else {
                if stack.isEmpty || stack.last != matching[char] {
                    return false
                }
                stack.popLast()
            }
        }

        return stack.isEmpty
    }
}